packages/cardpay-subgraph/src/mappings/register-reward-program-handler.ts

```diff
@@ -0,0 +1,33 @@
+import { RewardProgramRegistrationFee } from '../../generated/RewardProgram/RegisterRewardProgramHandler';
+import { log } from '@graphprotocol/graph-ts';
+
+import { RewardProgramRegistrationPayment } from '../../generated/schema';
+import { toChecksumAddress, makePrepaidCardPayment, makeToken, makeTransaction } from '../utils';
+
+export function handleRewardProgramRegistrationFee(event: RewardProgramRegistrationFee): void {
+  makeTransaction(event);
+  let prepaidCard = toChecksumAddress(event.params.prepaidCard);
+  let admin = toChecksumAddress(event.params.admin);
```

Comment:
you need to make sure to create an Account entity first for this admin if one doesn't already exist, or you'll get an index stopping error here.

Reply:
Got it.

Reply:
hmmm wait. what do you mean by "Create an Account"? Is this hidden within the action of bridging tokens?

Reply:
I mean create the Account entity. 
```js
let accountEntity = new Account(admin);
accountEntity.save();
```
It’s not good to assume that it will be there. Anytime we create an edge to an account entity in the subgraphs we perform this step. If an account already exists, then the graph node will do nothing. This is a good defensive strategy that prevents a breaking index. (And example might be that you need to debug something and want to index at a relatively high block height)

Reply:
It might not be a bad idea to create a `makeAccount()` function in the utils that does this, we do it a lot…